cfme/tests/candu/test_host_graph.py

```diff
@@ -233,18 +233,15 @@ def test_tagwise(provider, interval, graph_type, gp_by, candu_tag_vm, enable_can
     assert graph.is_displayed
 
     def refresh():
-        provider.refresh_provider_relationships()
-        host.capture_historical_data()
         provider.browser.refresh()
         view.wait_displayed(timeout='20s')
         view.options.fill(data)
 
     # wait, for specific vm tag data. It take time to reload metrics with specific vm tag.
-    wait_for(lambda: "London" in graph.all_legends,
-    delay=120, timeout=1500, fail_func=refresh)
+    wait_for(lambda: "London" in graph.all_legends, delay=60, timeout=3000, fail_func=refresh)
```

Comment:
In the PRT results, I'm seeing the activity spinner active on the page - is it possible this fail_func refresh is actually triggering it to re-load, and thus is never allowing the activity spinner to complete? 

Refresh is actually doing a fill, re-filling the form every iteration. Doesn't matter how long the timeout is if every time its re-trying the options fill and submit.

Reply:
It was my bad that i forgot to remove rhv related my tried. I increased timeout for rhv test. 
Yup, after browser refresh we need to refill data. need to select tag.